ARC-00XX: smartASA

aldur opened the pull request.

barnjamin committed:
adding first pass at royalties

barnjamin committed:
some notes

barnjamin committed:
Fmt

cusma committed:
Add checks on `AuthAddr`

barnjamin committed:
Merge pull request #1 from cusma/royalty

Add checks on `AuthAddr`

cusma committed:
Refining the `AuthAddr` checks.

barnjamin committed:
Merge pull request #2 from cusma/royalty

Refining the `AuthAddr` checks.

cusma committed:
Add the Royalty Collector role

barnjamin committed:
Merge pull request #3 from cusma/royalty

Add the Royalty Collector role

barnjamin committed:
tweaks to api

barnjamin committed:
Rename

barnjamin committed:
tweak name

barnjamin committed:
updates for offer/rescind

barnjamin committed:
add payment_asset option

barnjamin committed:
address enforcement group size issue

barnjamin committed:
Add details about global/local storage and metadata section describing how to store application id

cusma committed:
More permissive check on RekeyTo

cusma committed:
Royalty spec applies to secondary sales too

cusma committed:
Make ARC-18 Smart ASA compliant

cusma committed:
Minor fix

barnjamin committed:
Update ARCs/arc-0018.md

barnjamin committed:
Merge pull request #4 from cusma/royalty

Review

barnjamin committed:
adding royalty_asset_amount

barnjamin committed:
tweak abi

barnjamin committed:
prose for how to prevent royalty enforcement circumvention

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Fabrice Benhamouda <fbenhamo102@gmail.com>

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Fabrice Benhamouda <fbenhamo102@gmail.com>

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Fabrice Benhamouda <fbenhamo102@gmail.com>

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Fabrice Benhamouda <fbenhamo102@gmail.com>

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Fabrice Benhamouda <fbenhamo102@gmail.com>

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Fabrice Benhamouda <fbenhamo102@gmail.com>

barnjamin committed:
partial updates from comments

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Fabrice Benhamouda <fbenhamo102@gmail.com>

barnjamin committed:
remove create_nft, move circumvention techniques to security considerations, rename move to royalty_free_move

barnjamin committed:
Merge branch 'royalty' of github.com:barnjamin/ARCs into royalty

barnjamin committed:
Adding read only method for getting the policy

barnjamin committed:
Add get_offer method

barnjamin committed:
adding prose to describe how to update offer after transfer

barnjamin committed:
formatting and typos

barnjamin committed:
adding flow chart

barnjamin committed:
typo

barnjamin committed:
updating offer method to require any previous values be passed

barnjamin committed:
Adding example for updated offer

barnjamin committed:
add Adriano to contribs

barnjamin committed:
updating abi

barnjamin committed:
adding asset offer definition

barnjamin committed:
Update ARCs/arc-0018.md

barnjamin committed:
Update ARCs/arc-0018.md

barnjamin committed:
adding mention that royalty policy SHOULD be immutable

barnjamin committed:
make immutability a MUST

barnjamin committed:
rework based on Jason feedback

barnjamin committed:
clean up with new methods

barnjamin committed:
single set policy method

barnjamin committed:
mentioning of asset parameters, changing some wording, adding jasons gh

barnjamin committed:
add more metadata fields for convenience

barnjamin committed:
fix abi.json

barnjamin committed:
address comments

barnjamin committed:
immutable policy should

barnjamin committed:
more text in security about mutability

barnjamin committed:
remove extra fields

barnjamin committed:
clarify storage value

barnjamin committed:
clarification of 0 for payment asset

barnjamin committed:
clarify auth addr

barnjamin committed:
added missing from parameter

barnjamin committed:
Adding notes about royalty free move

barnjamin committed:
clarification on transfer needing royalty policy

barnjamin committed:
Adding back readonly methods

barnjamin committed:
specify that app should use fee pooling and standarize naming

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Simon Ingeson <44818+smonn@users.noreply.github.com>

barnjamin committed:
adding royalty enforcer admin

barnjamin committed:
sync abi with impl

barnjamin committed:
adding admin to prose

barnjamin committed:
Update ARCs/arc-0018.md

Co-authored-by: Simon Ingeson <44818+smonn@users.noreply.github.com>

barnjamin committed:
update abi to reflect longer name

barnjamin committed:
normalize arg naming/ordering

barnjamin committed:
names to shorter version

barnjamin committed:
adding rekey checked spec

ori-shem-tov committed:
Update arc-0001.md

replace `inner transaction` with `underlying transaction` to prevent confusion with AVM inner-transactions

barnjamin committed:
fixed name on get policy

aldur committed:
Initial commit

ori-shem-tov committed:
Update arc-0007.md to support transaction groups with only 1 transaction

this way dApps don't have to have special cases for single transactions and can always send a group to the wallet.

ori-shem-tov committed:
Update arc-0001.md

cusma commented on ARCs/arc-00xx.md:
```diff
@@ -0,0 +1,235 @@
+    { "type": "address", "name": "ManagerAddr" },
+    { "type": "address", "name": "ReserveAddr" },
+    { "type": "address", "name": "FreezeAddr" },
+    { "type": "address", "name": "ClawbackAddr" }
```
```suggestion
```

Maybe we should remove the `ClawbackAddr` both from `AssetCreate` and `AssetConfig` since the interface mandates that this can not be different from the `ApplicationAddress`.

aldur replied:
This is the `ClawbackAddr` of the Smart ASA, not of the underlying ASA (meaning I want to give you access to clawback _through_ the SC).

cusma replied:
Right! My bad!

barnjamin committed:
Adding denotation of optional/required on getters

barnjamin committed:
adding namespaced

aldur committed:
Update metadata specification

aldur committed:
Add non-normative sections

aldur committed:
Better links

aldur committed:
Fixes

aldur committed:
Cosmetic fixes

aldur committed:
Add rationale and backwards compatibility

aldur committed:
Add getters and opt in / close out security considerations

aldur committed:
Prose about handling opt in and close out

barnjamin committed:
change arc number

fabrice102 committed:
Merge pull request #89 from ori-shem-tov/arc-1-inner-to-underlying

Update arc-0001.md

fabrice102 committed:
Merge pull request #90 from ori-shem-tov/update-arc-7

Update arc-0007.md to support transaction groups with only 1 transaction

Commit:
add git page

Commit:
test it locally

aldur committed:
Add TODO

barnjamin committed:
Remove Typo

aldur committed:
Add getter for frozen status

aldur committed:
IsAssetFrozen -> IsAccountFrozen

aldur committed:
Remove link to outdated implementation

fabrice102 committed:
Merge pull request #94 from barnjamin/patch-2

Remove Typo

Commit:
fix https://github.com/algorandfoundation/ARCs/pull/93#issuecomment-1121034568

Commit:
rm _site and  EIP keyword

ori-shem-tov committed:
Update index.html

jasonpaulos committed:
Create ARC-25

jasonpaulos committed:
Remove unused header

jasonpaulos committed:
Fix typo

Co-authored-by: Ben Guidarelli <ben.guidarelli@gmail.com>

jasonpaulos committed:
Move method reference forward

ori-shem-tov committed:
Update index.html

ori-shem-tov committed:
Merge pull request #93 from lifecoder1988/main

add git page

ori-shem-tov committed:
Update arc-0001.md

ori-shem-tov committed:
Update arc-0002.md

ori-shem-tov committed:
Update arc-0003.md

ori-shem-tov committed:
Update arc-0004.md

ori-shem-tov committed:
Update arc-0005.md

ori-shem-tov committed:
Update arc-0006.md

ori-shem-tov committed:
Update arc-0007.md

ori-shem-tov committed:
Update arc-0008.md

ori-shem-tov committed:
Update arc-0009.md

ori-shem-tov committed:
Update arc-0010.md

ori-shem-tov committed:
Update arc-0011.md

ori-shem-tov committed:
Update arc-0019.md

ori-shem-tov committed:
Merge pull request #99 from ori-shem-tov/attach-discussions

Attach discussions

ori-shem-tov committed:
Update README.md

ori-shem-tov committed:
Merge pull request #100 from ori-shem-tov/link-page-readme

Link the github page in the README file

aldur committed:
Fix ABI array type (credits to @smonn)

aldur committed:
ABI to snake case

aldur committed:
ARC-20!

aldur committed:
Add prose for getters

aldur committed:
Add global freeze (set and get)

aldur committed:
Frozen accounts should not receive either

aldur committed:
Fix permissions specification for `asset_config`.

aldur committed:
Improve wording

aldur committed:
`get_meta_data_hash` -> `get_metadata_hash`

fabrice102 committed:
Create CNAME

ori-shem-tov committed:
Update index.html

ori-shem-tov committed:
Update _config.yml

ori-shem-tov committed:
Update README.md

ori-shem-tov committed:
Merge pull request #103 from ori-shem-tov/change-pages-layout

Change pages layout

jasonpaulos committed:
Fix typo

ori-shem-tov committed:
Merge pull request #96 from jasonpaulos/walletconnect-v1-arc

Create ARC-25: Algorand WalletConnect v1 API

aldur committed:
Fix header levels

aldur committed:
Add `circulating_supply`

ori-shem-tov committed:
added link to discussion

ori-shem-tov committed:
Update arc-0025.md

ori-shem-tov committed:
Update arc-0069.md

ori-shem-tov committed:
Merge pull request #105 from ori-shem-tov/linking-discussions

Linking discussions

aldur committed:
`asset_name` -> `name`

mxmauro committed:
Enforce required fields

fabrice102 committed:
Merge pull request #106 from mxmauro/patch-1

Enforce required fields in ARC-0069

barnjamin committed:
adding note about arc20

aldur committed:
Small changes

barnjamin committed:
Adding notes about different method signatures

fabrice102 committed:
Merge pull request #70 from barnjamin/royalty

ARC-0018 ASA Transfer Royalty Enforcement Specification

fabrice102 committed:
Update arc-0018.md

Add link to discussion

aldur committed:
Add full ABI specification

fabrice102 committed:
Last call for ARC-19

The main change is the clarification of the interaction with ARC-3
(there is a need for clients to bypass Asset Metadata validation
in that case.)

Other changes are mostly cosmetic. This includes Markdown fixes
(including text immediately after \`\`\` without newline
breaks markdown)

fabrice102 committed:
Merge branch 'main' of github.com:algorandfoundation/ARCs into arc19-final-call

aldur committed:
Update naming to match other getters

aldur committed:
Add missing methods to full spec

aldur committed:
`get_circulating_supply`

aldur committed:
Update wording, thanks for the feedbacks @robdmoore

aldur committed:
Add marketplaces to wording

aldur committed:
Improve wording

fabrice102 committed:
Merge pull request #116 from fabrice102/arc19-final-call

ARC-0019: final call

aldur committed:
Transfer should fail if asset is frozen

aldur committed:
Mark read-only methods

aldur committed:
Specify close-out

aldur committed:
Small fixes

aldur committed:
Improve security considerations, locate state and `default_frozen`

aldur committed:
Switch to a single configuration getter

aldur committed:
Merge branch 'algorandfoundation:main' into smartasa

aldur committed:
Add reference implementation
